src/components/spaces/canvas/index.js

```diff
@@ -222,10 +222,16 @@ export default class Canvas extends Component{
 
       let pathStatus = unconnectedStatus
       if (!withinSelectedRegion && (outputIsAncestor || inputIsDescendant)) {
+        const degree = outputIsAncestor ?
+          ancestors.findIndex(e => e.id === outputId) + 1
```

Comment:
So, after looking at this, I think this will only work for when you have just one metric selected, and it works in that case because then ancestors[0] === descendants[0] === the one selected metric So this loop cycles through and catches all the edges that either lead into the selected metric as degree zero or that lead out of the selected metric as degree zero. 

I think what you really want here is something like:
```
const isDegreeOne = _.some(selectedMetrics, ({id}) => [inputId, outputId].includes(id))
```